test/ng/httpSpec.js

```diff
@@ -781,6 +781,34 @@ describe('$http', function() {
         $httpBackend.flush();
       });
 
+      it('should expose a config object to header functions', function() {
+        var config = {
+          foo: 'Rewritten',
+          headers: {'Accept': function(config) {
+            return config.foo;
+          }}
+        };
+
+        $httpBackend.expect('GET', '/url', undefined, {Accept: 'Rewritten'}).respond('');
+        $http.get('/url', config);
+        $httpBackend.flush();
+      });
+
+      it('should not allow modifications to a config object in header functions', function() {
```

Comment:
this will also only "ignore" changes to root properties of the config object, so changes to child object properties won't be ignored :( but I don't think it's worth worrying about it until someone complains